### DISCUSSION TITLE
pnpm run failed,  many dependencies were not found. please

### DISCUSSION THREAD

**Discussion by CharlesJoyce** [pnpm/pnpm#4436] [Q&A]
**pnpm version:**
6.32.3

**Expected behavior:**
npm run dev success
pnpm run dev success

**Actual behavior:**
npm run dev success
pnpm run dev fail

pnpm i
pnpm run dev
> vue-cli-service serve

These dependencies were not found:

* @babel/runtime/helpers/assertThisInitialized in ./node_modules/video.js/dist/video.es.js
* @babel/runtime/helpers/construct in ./node_modules/video.js/dist/video.es.js
* @babel/runtime/helpers/extends in ./node_modules/video.js/dist/video.es.js
* @babel/runtime/helpers/inherits in ./node_modules/video.js/dist/video.es.js
* @babel/runtime/helpers/inheritsLoose in ./node_modules/video.js/dist/video.es.js
* @sentry/core in ./node_modules/@sentry/browser/esm/index.js, ./node_modules/@sentry/browser/esm/integrations/breadcrumbs.js and 1 other
* @sentry/hub in ./node_modules/@sentry/tracing/esm/utils.js
* @sentry/utils in ./node_modules/@sentry/browser/esm/index.js, ./node_modules/@sentry/browser/esm/integrations/globalhandlers.js and 5 others
* @vant/use in ./node_modules/vant/es/action-bar/ActionBar.js, ./node_modules/vant/es/action-bar-button/ActionBarButton.js and 12 others
* @videojs/vhs-utils/es/byte-helpers in ./node_modules/video.js/dist/video.es.js
* @videojs/vhs-utils/es/codecs.js in ./node_modules/video.js/dist/video.es.js
* @videojs/vhs-utils/es/containers in ./node_modules/video.js/dist/video.es.js
* @videojs/vhs-utils/es/id3-helpers in ./node_modules/video.js/dist/video.es.js
* @videojs/vhs-utils/es/media-types.js in ./node_modules/video.js/dist/video.es.js
* @videojs/vhs-utils/es/resolve-url.js in ./node_modules/video.js/dist/video.es.js
* @videojs/xhr in ./node_modules/video.js/dist/video.es.js
* @vue/devtools-api in ./node_modules/vue-router/dist/vue-router.esm-bundler.js, ./node_modules/vuex/dist/vuex.esm-browser.js
* @vue/runtime-dom in ./node_modules/vue/dist/vue.runtime.esm-bundler.js
* decode-uri-component in ./node_modules/query-string/index.js
* dijkstrajs in ./node_modules/qrcode/lib/core/segments.js
* encode-utf8 in ./node_modules/qrcode/lib/core/byte-data.js
* filter-obj in ./node_modules/query-string/index.js
* global/document in ./node_modules/video.js/dist/video.es.js
* global/window in ./node_modules/video.js/dist/video.es.js
* keycode in ./node_modules/video.js/dist/video.es.js
* lodash.assignin in ./node_modules/kwapi/dist/common/request.js, ./node_modules/kwapi/dist/interface/User.js
* lodash.foreach in ./node_modules/kwapi/dist/common/utils.js
* lodash.isfunction in ./node_modules/kwapi/dist/common/request.js
* lodash.isplainobject in ./node_modules/kwapi/dist/common/request.js
* m3u8-parser in ./node_modules/video.js/dist/video.es.js
* mitt in ./node_modules/vue3-eventbus/src/index.js
* mpd-parser in ./node_modules/video.js/dist/video.es.js
* mux.js/lib/tools/parse-sidx in ./node_modules/video.js/dist/video.es.js
* mux.js/lib/utils/clock in ./node_modules/video.js/dist/video.es.js
* regenerator-runtime/runtime.js in ./src/BLL/city.js, ./node_modules/.pnpm/cache-loader@4.1.0_webpack@4.46.0/node_modules/cache-loader/dist/cjs.js??ref--13-0!./node_modules/.pnpm/babel-loader@8.2.3_a58183228c2c7b1a132de84b2b6f9adf/node_modules/babel-loader/lib!./node_modules/.pnpm/cache-loader@4.1.0_webpack@4.46.0/node_modules/cache-loader/dist/cjs.js??ref--1-0!./node_modules/.pnpm/vue-loader@16.8.3_webpack@4.46.0/node_modules/vue-loader/dist??ref--1-1!./src/components/common/WorkerItem/new/index.vue?vue&type=script&lang=js and 35 others
* safe-json-parse/tuple in ./node_modules/video.js/dist/video.es.js
* sha1 in ./node_modules/kwapi/dist/common/utils.js
* side-channel in ./node_modules/qs/lib/stringify.js
* split-on-first in ./node_modules/query-string/index.js
* strict-uri-encode in ./node_modules/query-string/index.js
* tslib in ./node_modules/@sentry/browser/esm/index.js, ./node_modules/@sentry/browser/esm/integrations/globalhandlers.js and 6 others
* videojs-vtt.js in ./node_modules/video.js/dist/video.es.js
* whatwg-fetch in ./node_modules/isomorphic-fetch/fetch-npm-browserify.js

To install them, you can run: npm install --save @babel/runtime/helpers/assertThisInitialized @babel/runtime/helpers/construct @babel/runtime/helpers/extends @babel/runtime/helpers/inherits @babel/runtime/helpers/inheritsLoose @sentry/core @sentry/hub @sentry/utils @vant/use @videojs/vhs-utils/es/byte-helpers @videojs/vhs-utils/es/codecs.js @videojs/vhs-utils/es/containers @videojs/vhs-utils/es/id3-helpers @videojs/vhs-utils/es/media-types.js @videojs/vhs-utils/es/resolve-url.js @videojs/xhr @vue/devtools-api @vue/runtime-dom decode-uri-component dijkstrajs encode-utf8 filter-obj global/document global/window keycode lodash.assignin lodash.foreach lodash.isfunction lodash.isplainobject m3u8-parser mitt mpd-parser mux.js/lib/tools/parse-sidx mux.js/lib/utils/clock regenerator-runtime/runtime.js safe-json-parse/tuple sha1 side-channel split-on-first strict-uri-encode tslib videojs-vtt.js whatwg-fetch

**Additional information:**
node -v   v14.17.5
npm -v     8.5.0
Mac OS

**Comment by CharlesJoyce** [comment]
oh,my bad

**Reply by zkochan** [reply]
Have you solved it?

**Reply by CharlesJoyce** [reply]
Yes，thank you so much！

**Reply by hanzoz** [reply]
what is the solutions?

**Reply by clevertension** [reply]
same issue

**Reply by moonpie94** [reply]
what is the solutions?
